api/src/main/java/org/xnio/XnioProviderHolder.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ *
+ * Copyright 2014 Red Hat, Inc. and/or its affiliates, and individual
+ * contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.xnio;
+
+/**
+ * Holds an XnioProvider injected by a provider implementation.
+ * <p>
+ * Intended to be used with OSGi Declarative Services. No side
+ * effects when not running under OSGi.
+ * 
+ * @author Harald Wellmann
+ *
+ */
+public class XnioProviderHolder {
```

Comment:
I dont like the fact that this is not implementing the Holder pattern, so is somewhat confusing...